library/spit_results.py

```diff
@@ -173,15 +207,35 @@ def process_jboss_versions(host_vars):
             elif version.strip():
                 jboss_releases.append('Unknown-Release: ' + version)
 
-    if jboss_releases:
-        val['jboss.installed-versions'] = '; '.join(jboss_releases)
-        val['jboss.deploy-dates'] = '; '.join(deploy_dates)
+    running_versions = safe_ansible_property(host_vars,
+                                             JBOSS_RUNNING_VERSIONS,
+                                             'stdout')
+
+    def empty_output_message(val, name):
+        """Give the right error message for missing data.
+
+        For data that depends on having Java.
+
+        :param val: a value. Considered missing if falsey.
+        :param name: the thing we were searching for.
+        :returns: the val if true, otherwise a useful error message.
+        """
 
-    # The jboss role will not run if 'have_java' is false.
-    if not host_vars['have_java']:
-        val['jboss.installed-versions'] = 'N/A (java not found)'
-        val['jboss.deploy-dates'] = 'N/A (java not found)'
-        val[running_ver] = 'N/A (java not found)'
+        if val:
+            return val
+        if not host_vars['have_java']:
+            return 'N/A (java not found)'
+        return '({0} not found)'.format(name)
+
+    if JBOSS_INSTALLED_VERSIONS in fact_names:
+        val[JBOSS_INSTALLED_VERSIONS] = (
+            empty_output_message('; '.join(jboss_releases), 'jboss'))
+    if JBOSS_DEPLOY_DATES in fact_names:
+        val[JBOSS_DEPLOY_DATES] = (
+            empty_output_message('; '.join(deploy_dates), 'jboss'))
+    if JBOSS_RUNNING_VERSIONS in fact_names:
+        val[JBOSS_RUNNING_VERSIONS] = (
```

Comment:
The running_versions variable defined on line 210 seems to just be used inside this if block. I think it would be better to move its assignment inside the block to improve readability of the code.

Reply:
Done, thanks! I ended up just inlining the definition at the one place where it is used. What do you think?

Reply:
Even better.